common/event_emitter.go

```diff
@@ -97,13 +97,17 @@ type NavigationEvent struct {
 	err         error
 }
 
+type queue struct {
+	writeMutex sync.Mutex
+	write      []Event
+	readMutex  sync.Mutex
+	read       []Event
+}
+
 type eventHandler struct {
-	ctx           context.Context
-	ch            chan Event
-	queueMutex    sync.Mutex
-	queue         []Event
-	curQueueMutex sync.Mutex
-	curQueue      []Event
+	ctx   context.Context
+	ch    chan Event
+	queue *queue
```

Comment:
It is safer as a `pointer`. There is a `map` of `chan Event` -> `queue`, and this can be accessed at anytime during the runtime when an element calls `on` in `BaseEventEmitter`; e.g. `frame_session` calls `on` twice. The `queue` in the `map` and the `queue` in the `handler` need to be the same otherwise we could get out of order events and eventually deadlocks.